vlayer/ethereum/circuit/lib/src/token.nr

```diff
@@ -0,0 +1,40 @@
+use dep::ethereum_history_api::{account_with_storage::{get_account_with_storage, StorageWithinBlock}, misc::types::{Address, Bytes32}};
+use dep::std::unsafe::zeroed;
+
+global TOKEN_BALANCE_INDEX = 0;
+
+pub fn get_token_balance(token_address: Address, wallet_address: Address, block_number: Field) -> Field {
```

Comment:
I would propose following direction:
```rust
struct Token {
  balances_slot: ... // Field?
  address: Address
  ...
}
```
get_token_balance(token: Token, owner: Address) {
  ...
}
```

The direction with functions you proposed is inspiring thou, but I would consider doing it in more Noir/Rust idiomatic way:
```
impl Token {
  fn calculate_balance_slot(address: Address) -> u256
}
```

Finally, we need to go back to Leon'a prototype of DSL for calculating slots, to implement it. With that we will not needn storage_value_to_balance and get_account_with_storage.

Another question: What do we return? Field, u128, u256?
I propose to start with field and add ticket to implement u256 and swap.
Aztec says u256 is coming later this year, so for now we can make ineffective implementation and swap later.

Reply:
I've changed the implementation and now using struct and trait.